source/_docs/configuration/splitting_configuration.markdown

```diff
@@ -495,4 +495,98 @@ automation manual: !include_dir_merge_list automations/
 automation ui: !include automations.yaml
 ```
 
+## YAML Fragments
+
+Another option for making complex configuration files easier to read and maintain is the use of YAML fragments, or 'scraps', that are injected into your configuration. For example, let's say you have a few automations that are triggered by various sensors, but part of your conditions for those triggers is that you are home and the sun is up:
+
+`automations.yaml`
+```yaml
+condition:
+  - condition: state
+    entity_id: 'device_tracker.paulus'
+    state: 'home'
+  - condition: state
+    entity_id: sun.sun
+    state: 'above_horizon'
+```
+
+If you have to repeat these conditions in your automations YAML, the YAML quickly becomes 'noisy' and error-prone. To solve this, you can save these conditions in a separate 'scraps.yaml' file, and inject them into the configuration:
+
+`scraps.yaml`
+```yaml
+paulus_is_home_sun_is_up:
+  - condition: state
+    entity_id: 'device_tracker.paulus'
+    state: 'home'
+  - condition: state
+    entity_id: sun.sun
+    state: 'above_horizon'
+```
+You would then call this fragment into your automation:
+
+`automations.yaml`
+```yaml
+condition: ~scrap paulus_is_home_sun_is_up
```

Comment:
`[Tilde] [keyword]` is not valid YAML syntax. YAML defines custom tags for these purposes.